src/lib/y2network/widgets/gateway.rb

```diff
@@ -0,0 +1,44 @@
+Yast.import "IP"
+Yast.import "Popup"
+Yast.import "UI"
+
+require "cwm/common_widgets"
+
+module Y2Network
+  module Widgets
+    class Gateway < CWM::InputField
+      # @param route route object to get and store gateway value
+      def initialize(route)
+        textdomain "network"
+      end
+
+      def label
+        _("&Gateway")
+      end
+
+      def help
+        # TODO: original also does not have help
+        ""
+      end
+
+      def opt
+        [:hstretch]
+      end
+
+      def init
+        Yast::UI.ChangeWidget(Id(:gateway), :ValidChars, Yast::IP.ValidChars +"-")
+        # TODO: init from route object
+      end
+
+      def validate
+        return true if value == "-"
```

Comment:
hm. "-" is special char for sysconfig file ... but i don't think we need to support it in the UI.

Reply:
well, I do not plan to do backend or code modification with functionality. I just plan to modify UI and how it use backend. And now it still uses "-" in UI. I tested it also manually. In fact it is default value there.

Reply:
for current code see:

https://github.com/yast/yast-network/blob/master/src/include/network/services/routing.rb#L191
https://github.com/yast/yast-network/blob/master/src/include/network/services/routing.rb#L220
https://github.com/yast/yast-network/blob/master/src/include/network/services/routing.rb#L152

Reply:
i know it uses it ... but its usual meaning is "unspecified". We can use some user more friendly way how to say that.

Reply:
yes, fine for me. We will see also how our model define it.

Reply:
and hopefully also writting help text which mentions special values can help.